package maidsafe.lifestuff;

message PublicContact {
  required bytes public_username = 1;
  required bytes mpid_name = 2;
  required bytes mmid_name = 3;
  required bytes mpid_public_key = 4;
  required bytes mmid_public_key = 5;
  required uint32 status = 6;
  required uint32 rank = 7;
  required uint32 last_contact = 8;
  optional bytes pointer_to_info = 9;
}

message PersonalDetails {
  optional bytes full_name = 1;
  optional bytes phone_number = 2;
  optional bytes birthday = 3;
  optional bytes gender = 4;
  optional int32 language = 5;
  optional bytes city = 6;
  optional int32 country = 7;
}

message DataAtlas {
  optional bytes unique_user_id = 1;
  optional bytes root_parent_id = 2;
  optional bytes serialised_keyring = 3;
  optional bytes serialised_selectables = 4;
  optional bytes timestamp = 5;
  repeated PublicContact contacts = 6;
}

